environments/openshift/prometheus.libsonnet

```diff
@@ -0,0 +1,275 @@
+        resourceRequests: {},
+        remoteWrite: [
+          {
+            url: $._config.ams.remoteWriteProxy,
+            writeRelabelConfigs: [
+              {
+                sourceLabels: ['__name__'],
+                regex: 'subscription_labels',
+                action: 'keep',
+              },
+            ],
+          },
+        ],
+      },
+    },
+  },
+
+  prometheusAms+:: {
+    service:
+      local service = k.core.v1.service;
+      local servicePort = k.core.v1.service.mixin.spec.portsType;
+
+      local prometheusPort = servicePort.newNamed('web', 9090, 'web');
+
+      service.new('prometheus-' + $._config.ams.prometheus.name, { 'app.kubernetes.io/name': 'prometheus', prometheus: $._config.ams.prometheus.name }, prometheusPort) +
+      service.mixin.spec.withSessionAffinity('ClientIP') +
+      service.mixin.metadata.withNamespace($._config.namespace) +
+      service.mixin.metadata.withLabels({ prometheus: $._config.ams.prometheus.name }),
+    [if $._config.ams.prometheus.rules != null && $._config.ams.prometheus.rules != {} then 'rules']:
+      {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'PrometheusRule',
+        metadata: {
+          labels: {
+            prometheus: $._config.ams.prometheus.name,
+            role: 'alert-rules',
+          },
+          name: 'prometheus-' + $._config.ams.prometheus.name + '-rules',
+          namespace: $._config.namespace,
+        },
+        spec: {
+          groups: $._config.ams.prometheus.rules.groups,
+        },
+      },
+    prometheus:
+      local statefulSet = k.apps.v1.statefulSet;
+      local container = statefulSet.mixin.spec.template.spec.containersType;
+      local resourceRequirements = container.mixin.resourcesType;
+      local selector = statefulSet.mixin.spec.selectorType;
+      local deployment = k.apps.v1.deployment;
+
+      local resources =
+        resourceRequirements.new() +
+        resourceRequirements.withLimits({
+          cpu: '${PROMETHEUS_AMS_CPU_LIMIT}',
+          memory: '${PROMETHEUS_AMS_MEMORY_LIMIT}',
+        }) +
+        resourceRequirements.withRequests({
+          cpu: '${PROMETHEUS_AMS_CPU_REQUEST}',
+          memory: '${PROMETHEUS_AMS_MEMORY_REQUEST}',
+        });
+
+      {
+        apiVersion: 'monitoring.coreos.com/v1',
+        kind: 'Prometheus',
+        metadata: {
+          name: $._config.ams.prometheus.name,
+          namespace: $._config.namespace,
+          labels: {
+            prometheus: $._config.ams.prometheus.name,
+          },
+        },
+        spec: {
+          replicas: $._config.ams.prometheus.replicas,
+          version: '${PROMETHEUS_AMS_IMAGE_TAG}',
+          baseImage: '${PROMETHEUS_AMS_IMAGE}',
+          securityContext: {},
+          serviceAccount: 'prometheus-telemeter',
+          serviceAccountName: 'prometheus-telemeter',
+          serviceMonitorSelector: selector.withMatchLabels({
+            'k8s-app': 'prometheus-ams',
```

Comment:
we no longer use this label name and instead try to use `app.kubernetes.io/name` consistently